trace-sdk/src/main/java/io/bitrise/trace/TraceSdk.java

```diff
@@ -69,11 +75,29 @@ public synchronized static void init(@NonNull final Context context) {
             initDataCollection(context);
             initLifeCycleListener(context);
             initNetworkTracing();
+            TraceLog.i(String.format(LogMessageConstants.TRACE_DEBUG_FLAG_STATUS, DEBUG_ENABLED));
         } else {
             TraceLog.e(new TraceException.TraceConfigNotInitialisedException());
         }
     }
 
+    /**
+     * Flag to enable TraceSdk into debug mode. Currently, this will add more debug log messages.
+     *
+     * @param debugEnabled boolean value to enable or disable debug mode in the TraceSdk.
+     */
+    public synchronized static void setDebugEnabled(final boolean debugEnabled) {
+        DEBUG_ENABLED = debugEnabled;
+        TraceLog.i(String.format(LogMessageConstants.TRACE_DEBUG_FLAG_STATUS, DEBUG_ENABLED));
```

Comment:
This will make logging silent implicitly, if the logger is not initialised yet. See [here](https://github.com/bitrise-io/trace-android-sdk/blob/main/trace-sdk/src/main/java/io/bitrise/trace/utils/log/TraceLog.java#L128). I think we have to mention this in the readme, to set this enabled after the SDK is initialised, otherwise log messages will be silent.

Reply:
I added the following to the three places people will see info about it (the flag variable, the setter, and the readme):
> Note: we do not log anything if your app is in release mode. To see these log messages your app should also be in debug mode.

Does that look okay?

Reply:
I am not sure if that is correct, because developers can have such configuration, that their release version of the app does not strip out debug level logs with proguard.

But what I mentioned above in the comment is a different thing. If we call setDebugEnabled() method before the TraceSDK.init() is called, TraceLog will create an instance of SilentLogger, so no logs will be seen on Logcat.

Reply:
Hmm. Perhaps both are valid to mention?

How does this sound (figured i'd get your feedback before pushing the code):
> The TraceSdk has a debug mode - currently this will mean more debug level log messages. 
We do not log anything if your app is not in debug mode unless you have customised this with something like proguard. 
You also need to ensure that the TraceSdk has been initialised before setting the debug enabled mode.

Reply:
> We do not log anything if your app is not in debug mode unless you have customised this with something like proguard.

This depends on the state of minify, and of course the configuration of the rules. I believe the simplest solution is always the best: Please note if minify is enabled it can affect these logs, and they can be stripped out depending on your configuration.

Reply:
What do you mean by the simplest solution is always best ?

> The TraceSdk has a debug mode - currently this will mean more debug level log messages.
Please note if you are not using a debug build, and or minify is enabled it can affect these logs, and they can be stripped out depending on your configuration
You also need to ensure that the TraceSdk has been initialised before setting the debug enabled mode.

Does this work?

Reply:
Exactly, thanks!

Reply:
Perfect - will update the code!

Reply:
Thank you for your help on this btw - sometimes it takes a few messages to get the wording right - these things are really important :)